include/deal.II/sundials/ida_interface.h

```diff
@@ -0,0 +1,476 @@
+                 const double &min_step_size = 1e-6,
+                 const double &abs_tol = 1e-6,
+                 const double &rel_tol = 1e-5,
+                 const unsigned int &max_order = 5,
+                 const double &output_period = .1,
+                 const bool &ignore_algebraic_terms_for_errors = true,
+                 const std::string &ic_type = "use_y_diff",
+                 const std::string &reset_type = "use_y_dot",
+                 const double &ic_alpha = .33,
+                 const unsigned int &ic_max_iter = 5,
+                 const unsigned int &max_non_linear_iterations = 10,
+                 const bool &verbose = true,
+                 const bool &use_local_tolerances = false);
+
+    /**
+     * House cleaning.
+     */
+    ~IDAInterface();
+
+    /**
+     * Add all internal parameters to the given ParameterHandler object. When
+     * the paramaters are parsed from a file, the internal parameters are automatically
+     * updated.
+     *
+     * The following parameters are declared:
+     *
+     * @code
+     * set Absolute error tolerance                      = 0.000001
+     * set Final time                                    = 1.000000
+     * set Ignore algebraic terms for error computations = true
+     * set Initial condition Newton max iterations       = 5
+     * set Initial condition Newton parameter            = 0.330000
+     * set Initial condition type                        = use_y_diff
+     * set Initial condition type after restart          = use_y_dot
+     * set Initial step size                             = 0.000100
+     * set Initial time                                  = 0.000000
+     * set Maximum number of nonlinear iterations        = 10
+     * set Maximum order of BDF                          = 5
+     * set Min step size                                 = 0.000001
+     * set Relative error tolerance                      = 0.000010
+     * set Show output of time steps                     = true
+     * set Time units between each output                = 0.05
+     * set Use local tolerances                          = false
+     * @endcode
+     *
+     * These are one-to-one with the options you can pass at construction time.
+     *
+     * Those options are set as default value in the ParameterHandler object
+     * `prm`.
+     */
+    virtual void add_parameters(ParameterHandler &prm);
+
+    /**
+     * Evolve. This function returns the final number of steps.
+     */
+    unsigned int solve_dae(VectorType &solution,
+                           VectorType &solution_dot);
+
+    /**
+     * Clear internal memory, and start with clean objects. This function is
+     * called when the simulation start and when the user return true to a call
+     * to solver_should_restart()
+     */
+    void reset_dae(const double t,
+                   VectorType &y,
+                   VectorType &yp,
+                   double h,
+                   bool first_step);
```

Comment:
can you document what `v` and `yp` are set to (I suspect they are outputs?) and make `h` and `first_step` `const`.